File: MajorityElement.py
def majorityElement(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        a = {}

        for c in nums:
            if c in a:
                a[c] += 1
                if a[c] > len(nums)/2:
                    return c
            else:
                a[c] = 1
            

File: MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
        
        int majEl;
        
        int size = nums.size();
        
        int count = 0;
        
        for(int i = 0; i < size; i++)
        {
            if(count == 0)
            {
                majEl = nums[i];
                count = 1;
            }
            else
            {
                if(majEl == nums[i])
                {
                    count = count + 1;
                }

                else
                {
                    count = count - 1;
                }
            }
            
            
        }
        
        return majEl;
    }
};